app/script/calling/VideoGridRepository.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.calling = z.calling || {};
+
+z.calling.VideoGridRepository = (() => {
+  return class VideoGridRepository {
+    /**
+     * Construct an new VideoGridRepository.
+     * @param {MediaRepository} mediaRepository - Repository for the media streams
+     * @param {CallingRepository} callingRepository - Repository for the calls
+     */
+    constructor(mediaRepository, callingRepository) {
+      const streamHandler = mediaRepository.stream_handler;
+      const streamsInfo = streamHandler.remoteMediaStreamInfoIndex.video;
+      const localMediaStream = streamHandler.localMediaStream;
+      const selfStreamState = callingRepository.selfStreamState;
+      const calls = callingRepository.calls;
+      this.grid = ko.observableArray([0, 0, 0, 0]);
+      this.selfId = ko.observable(null);
```

Comment:
To be consistent with the rest of the code base we should use `ko.observable(undefined)` here.

Reply:
As discussed before, I am strongly of the opinion that ```undefined``` shouldn't be set explicitly if not absolutely necessary. 
I can change this occurrence now (to be consistent), but would really consider changing this behaviour throughout the codebase in the near future.